sanitylib: Adding generation of json format with more details

yerabolu opened the pull request:
Generate JSON format with more details on test, platform and
environment. JSON format co-exists with the other formats.

Signed-off-by: Spoorthy Priya Yerabolu <spoorthy.priya.yerabolu@intel.com>

nashif commented on scripts/sanity_chk/sanitylib.py:
```diff
@@ -3097,6 +3099,51 @@ def csv_report(self, filename):
+                           "platform": instance.platform.name,
+                           "handler": instance.platform.simulation,
+                           "status": instance.status,
+                           "toolchain": TestSuite.get_toolchain(),
```
this is going to be a global for all tests, no need to have this per item. We should look into splitting data into three parts:
- environment: things that are detected or coming from the execution env. such as toolchain, the OS we are running on, the version we are testing etc.
- testcases (mostly static data): Data we have from the testcase definition, not all data is relevant however, because we have most of that already well defined in yaml files, for example the fields used for filtering are not that useful, however the result of the filter (coming from apply_filters) is more significant... 
- the data that comes out of running the tests, here the one major thing I am missing is the log and output of the tests. This should be useful for both fail and pass and might result in a huge json file, but it is going to be a resource that can be used to validate the results after the fact if something went wrong. Other runtime data can be collected here.

If we have the data above, we should be able to take the json file and generate any kind of test reports with any format...

yerabolu committed:
sanitylib: Adding generation of json format with more details

Generate JSON format with more details on test, platform and
environment. JSON format co-exists with the other formats.

Signed-off-by: Spoorthy Priya Yerabolu <spoorthy.priya.yerabolu@intel.com>

nashif commented:
i guess we can close this
